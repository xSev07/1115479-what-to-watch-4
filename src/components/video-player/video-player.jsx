import React from "react";

const VideoPlayer = React.forwardRef((props, ref) => {
  return (
    <video
      ref={ref}
      width="280"
      height="175"
    />
  );
});

VideoPlayer.displayName = `VideoPlayer`;

export default VideoPlayer;
