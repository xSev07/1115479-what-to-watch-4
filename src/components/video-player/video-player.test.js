import React from "react";
import renderer from "react-test-renderer";
import {VideoPlayer} from "./video-player";

const innerComponent = <video/>;

describe(`Should VideoPlayer render correctly`, () => {
  it(`should render with start parameters`, () => {
    const tree = renderer
      .create(
          <VideoPlayer
            isPlaying={false}
            togglerPosition={0}
            elapsedTime={`1:30:00`}
            onPlayButtonClick={() => {}}
            onFullscreenButtonClick={() => {}}
            onExitButtonClick={() => {}}
          >
            {innerComponent}
          </VideoPlayer>, {
            createNodeMock: () => {
              return {};
            }
          }
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should render with custom parameters`, () => {
    const tree = renderer
      .create(
          <VideoPlayer
            isPlaying={true}
            togglerPosition={30}
            elapsedTime={`1:10:00`}
            onPlayButtonClick={() => {}}
            onFullscreenButtonClick={() => {}}
            onExitButtonClick={() => {}}
          >
            {innerComponent}
          </VideoPlayer>, {
            createNodeMock: () => {
              return {};
            }
          }
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
