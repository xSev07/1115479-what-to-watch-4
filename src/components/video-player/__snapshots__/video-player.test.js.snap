// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should VideoPlayer render correctly should render preview video player 1`] = `
<video
  height="175"
  width="280"
/>
`;
